docs/site/OpenAPI-generator.md

```diff
@@ -135,10 +137,148 @@ import {NewPet} from './new-pet.model.ts';
+      operationId: perform-search
+      parameters:
+        // ...
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  additionalProperties:
+                    type: object
+        '404':
+          description: No matching record found for the given criteria.
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                criteria:
+                  description: >-
+                    Uses Lucene Query Syntax in the format of
+                    propertyName:value, propertyName:[num1 TO num2] and date
+                    range format: propertyName:[yyyyMMdd TO yyyyMMdd]. In the
+                    response please see the 'docs' element which has the list of
+                    record objects. Each record structure would consist of all
+                    the fields and their corresponding values.
+                  type: string
+                  default: '*:*'
+                start:
+                  description: Starting record number. Default value is 0.
+                  type: integer
+                  default: 0
+                rows:
+                  description: >-
+                    Specify number of rows to be returned. If you run the search
+                    with default values, in the response you will see 'numFound'
+                    attribute which will tell the number of records available in
+                    the dataset.
+                  type: integer
+                  default: 100
+              required:
+                - criteria
+```
+
+Without `--promote-anonymous-schemas`, no separate files are generated for
+anonymous schemas. The controller class uses inline TypeScript type literals as
+shown below.
+
+{% include code-caption.html content="src/controllers/search.controller.ts" %}
+
+```ts
+@operation('post', '/{dataset}/{version}/records')
+  async performSearch(
+    @requestBody()
+    body: {
+      criteria: string;
+      start?: number;
+      rows?: number;
+    },
+    @param({name: 'version', in: 'path'}) version: string,
+    @param({name: 'dataset', in: 'path'}) dataset: string,
+  ): Promise<
+    {
+      [additionalProperty: string]: {};
+    }[]
+  > {
+    throw new Error('Not implemented');
+  }
+```
+
+On contrast, ff `lb4 openapi --promote-anonymous-schemas` is used, two
+additional model files are generated:
+
+{% include code-caption.html content="src/models/perform-search-body.model.ts" %}
+
+```ts
+/* tslint:disable:no-any */
+import {model, property} from '@loopback/repository';
+
+/**
+ * The model class is generated from OpenAPI schema - performSearchBody
+ * performSearchBody
+ */
+@model({name: 'performSearchBody'})
+export class PerformSearchBody {
```

Comment:
Should we include "Request" in the name, to distinguish from "Response" body? I.e. `PerformSearchRequestBody`. (PerformSearchResponseBody is already used later in this file).